src/test/kotlin/com/terraformation/backend/accelerator/db/ParticipantProjectSpeciesServiceTest.kt

```diff
@@ -0,0 +1,120 @@
+package com.terraformation.backend.accelerator.db
+
+import com.terraformation.backend.RunsAsUser
+import com.terraformation.backend.TestClock
+import com.terraformation.backend.TestEventPublisher
+import com.terraformation.backend.accelerator.ParticipantProjectSpeciesService
+import com.terraformation.backend.accelerator.model.NewParticipantProjectSpeciesModel
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.db.DatabaseTest
+import com.terraformation.backend.db.accelerator.DeliverableType
+import com.terraformation.backend.db.accelerator.SubmissionStatus
+import com.terraformation.backend.db.accelerator.tables.pojos.SubmissionsRow
+import com.terraformation.backend.mockUser
+import io.mockk.every
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+class ParticipantProjectSpeciesServiceTest : DatabaseTest(), RunsAsUser {
+  override val user = mockUser()
+
+  private val clock = TestClock()
+  private val eventPublisher = TestEventPublisher()
+
+  private val service: ParticipantProjectSpeciesService by lazy {
+    ParticipantProjectSpeciesService(
+        ParticipantProjectSpeciesStore(dslContext, participantProjectSpeciesDao, projectsDao),
+        SubmissionStore(clock, dslContext, eventPublisher))
+  }
+
+  @BeforeEach
+  fun setUp() {
+    insertUser()
+    insertOrganization()
+
+    every { user.canCreateParticipantProjectSpecies(any()) } returns true
+    every { user.canCreateSubmission(any()) } returns true
+    every { user.canReadProject(any()) } returns true
+    every { user.canReadProjectDeliverables(any()) } returns true
+  }
+
+  @Nested
+  inner class Create {
+    @Test
+    fun `creates a submission for the project and deliverable if one does not exist for the active module when a species is added to a project`() {
```

Comment:
Maybe worth a test to make sure it handles cases where there already is a deliverable.